### PR TITLE
PERF-2690 deploy mongos latency workload.

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

**Comment by jimoleary** [comment]
[4.4 Atlas Patch](https://spruce.mongodb.com/version/61938731a4cf473e7033accd/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)
[5.0 Atlas Patch](https://spruce.mongodb.com/version/619387de61837d1e387d5975/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) (I expect some of these tasks to fail)

File: src/workloads/issues/MongosLatency.yml [hunk 1]
```diff
@@ -0,0 +1,233 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/product-perf"
+Description: |
+  This workload was created to reproduce the issue described in HELP-25757 / SERVER-58997. The goal
```
Comment (brad-devlugt) [inline src/workloads/issues/MongosLatency.yml:4]: Should we switch to only reference the SERVER ticket here, since it is the public project?

**Review by brad-devlugt — APPROVED** [review]
LGTM

File: src/workloads/issues/MongosLatency.yml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  is to induce periods of latency in the MongoS processes that are not caused by the MongoD processes.
+
+  The repo comprises of:
```
Comment (dalyd) [inline src/workloads/issues/MongosLatency.yml:7]: s/repo/repro/ or even better reproduction.

File: src/workloads/issues/MongosLatency.yml [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    QueryOptions:
+      minPoolSize: 1
+      maxPoolSize: 10000
+
+GlobalDefaults:
+  DbName: &db test
+  Collection: &col Collection0
+  FullName: &fullname test.Collection0
+
+  ClassId: &classId {^ObjectId: {^FormatString: {"format": "c%04x%s", "withArgs":[{^RandomInt: {min: 0, max: 200}}, {^RandomString: {length: 19, alphabet: b}}]}}}
```
Comment (dalyd) [inline src/workloads/issues/MongosLatency.yml:29]: There's go to be a better way to do this. The alphabet has only one character in it, so this is making the string "bbbbbbbbbbbbbbbbbbb" but generating a fair number of random values along the way. Did you mean for this to be a string of b's?

File: src/workloads/issues/MongosLatency.yml [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+  StudentId: &studentId {^ObjectId: {^FormatString: {"format": "%04x%s", "withArgs": [{^RandomInt: {min: 0, max: 30}}, {^RandomString: {length: 20, alphabet: b}}]}}}
```
Comment (dalyd) [inline src/workloads/issues/MongosLatency.yml:30]: Same comment as the previous. This is going to make a string of bs.

File: src/workloads/issues/MongosLatency.yml (hunk 1 above)
Comment (jimoleary) [reply src/workloads/issues/MongosLatency.yml:4]: good idea, I'll update.

**Review by dalyd — CHANGES_REQUESTED** [review]

File: src/workloads/issues/MongosLatency.yml (hunk 2 above)
Comment (jimoleary) [reply src/workloads/issues/MongosLatency.yml:7]: :+1:

File: src/workloads/issues/MongosLatency.yml (hunk 3 above)
Comment (jimoleary) [reply src/workloads/issues/MongosLatency.yml:29]: Yeah, I think I was just slavishly copying the source repo (although it appends a). I'll put in the explicit strings for now.

**Review by dalyd — APPROVED** [review]

**Comment by jimoleary** [comment]
evergreen merge

**Pull request merged**